crates/ruff_linter/src/rules/flake8_quotes/rules/avoidable_escaped_quote.rs

```diff
@@ -210,6 +211,83 @@ impl Visitor<'_> for AvoidableEscapedQuoteChecker<'_> {
 
         walk_f_string(self, f_string);
     }
+
+    fn visit_t_string(&mut self, t_string: &'_ ast::TString) {
+        // If the target version doesn't support PEP 701, skip this entire f-string if it contains
+        // any string literal in any of the expression element. For example:
+        //
+        // ```python
+        // f"\"foo\" {'nested'}"
+        // ```
+        //
+        // If we try to fix the above example, the outer and inner quote will be the same which is
+        // invalid for any Python version before 3.12:
+        //
+        // ```python
+        // f'"foo" {'nested'}"
+        // ```
+        //
+        // Note that this check needs to be done globally to ignore the entire f-string. It is
+        // implicitly global in that we avoid recursing into this f-string if this is the case.
+        if !self.supports_pep701 {
```

Comment:
I don't think this is necessary for t-strings

The examples also need updating and it would be nice if we could share some of the logic with f-strings